pandas/core/algorithms.py

```diff
@@ -2214,7 +2214,9 @@ def _sort_mixed(values):
     return np.concatenate([nums, np.asarray(strs, dtype=object)])
 
 
-def _sort_tuples(values: np.ndarray[tuple]):
+# pandas\core\algorithms.py:2217: error: "ndarray" expects no type arguments,
+# but 1 given  [type-arg]
+def _sort_tuples(values: np.ndarray[tuple]):  # type: ignore[type-arg]
```

Comment:
IIUC np.ndarray takes annotations of the form `ndarray[shape, dtype]`, not sure that `tuple` make sense in this context

Reply:
ndarray doesn't take type arguments (is not a generic type). so this just needs to be changed to just `np.ndarray` on master.

https://github.com/numpy/numpy/blob/a74789081fd9753c5e5637c404523b8f47b55dcc/numpy/__init__.pyi#L1478

Reply:
opened #39792